src/resolve-environment-action.ts

```diff
@@ -0,0 +1,88 @@
+import * as core from "@actions/core";
+
+import {
+  createStatusReportBase,
+  getOptionalInput,
+  getRequiredInput,
+  getTemporaryDirectory,
+  sendStatusReport,
+} from "./actions-util";
+import { getGitHubVersion } from "./api-client";
+import * as configUtils from "./config-utils";
+import { Language, resolveAlias } from "./languages";
+import { getActionsLogger } from "./logging";
+import { runResolveBuildEnvironment } from "./resolve-environment";
+import { checkForTimeout, checkGitHubVersionInRange, wrapError } from "./util";
+import { validateWorkflow } from "./workflow";
+
+const ACTION_NAME = "resolve-environment";
+
+async function run() {
+  const startedAt = new Date();
+  const logger = getActionsLogger();
+  const language: Language = resolveAlias(getRequiredInput("language"));
+
+  try {
+    const workflowErrors = await validateWorkflow(logger);
+
+    if (
+      !(await sendStatusReport(
+        await createStatusReportBase(
+          ACTION_NAME,
+          "starting",
+          startedAt,
+          workflowErrors
+        )
+      ))
+    ) {
+      return;
+    }
+
+    const gitHubVersion = await getGitHubVersion();
+    checkGitHubVersionInRange(gitHubVersion, logger);
+
+    const config = await configUtils.getConfig(getTemporaryDirectory(), logger);
+    if (config === undefined) {
+      throw new Error(
+        "Config file could not be found at expected location. Has the 'init' action been called?"
+      );
+    }
+
+    const workingDirectory = getOptionalInput("working-directory");
+    const result = await runResolveBuildEnvironment(
+      config.codeQLCmd,
+      logger,
+      workingDirectory,
+      language
+    );
+    core.setOutput("environment", result);
+  } catch (unwrappedError) {
+    const error = wrapError(unwrappedError);
+    core.setFailed(error.message);
+    await sendStatusReport(
+      await createStatusReportBase(
+        ACTION_NAME,
+        "aborted",
```

Comment:
We use `aborted` for cases where we didn't manage to download the CodeQL CLI, and `getActionsStatus` elsewhere.  This lets us distinguish between user errors and non-user errors in telemetry.

```suggestion
        getActionsStatus(error),
```